Expand contributing docs (#15208)
conda/conda: CONTRIBUTING.md

### Contributing your changes to conda

> Here are the high level steps you need to take to contribute to conda:

Reviewer: ```suggestion
Here are steps you need to take to contribute to conda:
```

   "Fork" button on [https://github.com/conda/conda](https://github.com/conda/conda) and follow GitHub's
   instructions.
4. Work on your proposed solution. [Visit this page if you need help getting your development environment setup][development-environment]

> 5. When you are ready to submit a change, create a new pull request so that we
>    can merge your changes to our repository.
> 

Reviewer: ```suggestion
5. When you are ready to submit a change, create a pull request.

```

> #### Getting started with your first contribution

Reviewer: ```suggestion
#### Your first contribution
```

For your first contribution, we recommend:

> - Looking for issues labeled with `good first issue`
> - Starting with small changes like documentation or minor bug fixes
> - Reading through existing code to understand the project structure and our
>   patterns
> - Asking questions if you need help issues, on the
>   [community chat channels](https://conda.zulip.com), or on the
>   [conda discourse forum](https://conda.discourse.group)
> 

Reviewer: ```suggestion
- Look for issues labeled with `good first issue`
- Start with small changes like documentation or minor bug fixes
- Read through existing code to understand the project structure and our
  patterns
- Ask questions if you need help, on the
  [community chat channels](https://conda.zulip.com), or on the
  [conda discourse forum](https://conda.discourse.group)

```

#### Change process

In order to maintain proper tracking and visibility, make sure that there are
issues for adding features, fixing bugs, and major refactoring work. When
creating a new issue, make sure to use the issue template.

When making a change, try to scope changes to what can be accomplished in less
than a day of work with minimal code changes to facilitate rapid code
reviews—typically fewer than 200 lines of code.

> For each change, create a new local branch. Keep your changes focused on a
> single issue to make reviews easier.

Reviewer: ```suggestion
Create a new local branch for each change. Keep your changes focused on a
single issue to make reviews easier.
```

#### Pull Requests

##### Pre-submission Steps

1. Install and run pre-commit hooks
2. Make sure all tests are passing
3. Rebase changes on main if other changes were merged during development
4. Use the Pull Request template if one exists for the project
5. Self-review your code with fresh eyes before requesting external review
6. Draft pull requests are acceptable for early feedback

The Pull Request title should clearly explain the contents of the change. Like
the first line of a commit message, it should be a maximum of 50 characters,

> written in an imperative mood, and have no ending period.

Reviewer: ```suggestion
written in an imperative mood, and have no ending period
```
;-)

##### Review requirements

Most code changes require one reviewer from the conda-maintainers team. When
you submit a Pull Request, that group will automatically be assigned as the
reviewer.

##### What to expect during code review

> The primary goal of code review is ensuring future maintainability of
> integrated code. Reviews will focus on:

Reviewer: ```suggestion
The primary goal of code review is to ensure maintainability of
integrated code. Reviews will focus on:
```

- Code correctness and logic
- Maintainability and readability

> - Consistency with project patterns and best practices

Reviewer: ```suggestion
- Consistency with project patterns and practices
```
What is best ;-)

The Pull Request title should clearly explain the contents of the change. Like
the first line of a commit message, it should be a maximum of 50 characters,
written in an imperative mood, and have no ending period. It should stay
updated if the Pull Request is further refined so that if the Pull Request is
squashed and merged, the commit message reflects what was delivered.

##### Review Requirements

Standard Review: Most code changes require one reviewer from the
conda-maintainers team. Directly request a review from the person you
previously identified. If you paired with them during development, continuous
review counts as this requirement.

Second Review: Required only when the code author or the first reviewer feels
like it is necessary to get another set of eyes on a proposed change. In this
case, they add someone specific through GitHub’s Request Review feature.
Normally they should also put a comment to the person on what they want the
person to look for.

#### Code review process

The primary goal of code review is ensuring future maintainability of
integrated code. Reviews should not focus on:
- Code formatting (handled by pre-commit hooks)

> - Spelling (handled by automated checks)

Reviewer 2: No, we don't. I don't think this is a big enough problem to warrant us adding something like this. I could also see this become very annoying because we would have to add lots of exceptions to it (e.g. conda, repodata, subdir, etc.)

#### In links

Links should use the same capitalization conventions as sentences. Because the conda docs currently use reStructuredText (RST) as a markup language, and [RST does not support nested inline markup](https://docutils.sourceforge.io/FAQ.html#is-nested-inline-markup-possible), documentation writers should avoid using code backtick formatting inside links.

## Getting help

If you need help with your contribution:

- Comment on the issue you're working on
- Join our [community chat channels](https://conda.zulip.com)

> - Ask questions on our [conda discourse forum](https://conda.discourse.group)

Reviewer: Repeats L75, may be ok

Author: Thanks @dholth, deduplicated!
